<!--Launch file for nodelets-->
<!--see readme section on launch files for details on how to use file-->
<!--maintainer email="nxgamarra@gmail.com">Nikolas Gamarra</maintainer-->
<launch>

<!--Load settings as parameters from .yaml-->
  <group ns="settings">
    <rosparam command="load" file="$(find point_cloud_processing)/launch/point_cloud_settings.yaml" />
  </group>

  <node name="poin_cloud_processor1" pkg="point_cloud_processing" type="point_cloud_processing_node" output="screen">
    <param name="subscriber" value="/camera/depth/points"/>
    <param name="publisher" value="point_cloud_A"/>
    <param name="mode" value="passThrough"/>
    <param name="msgSubscriber" value="zzzzzzzzz"/>
  </node>

  <node name="poin_cloud_processor2" pkg="point_cloud_processing" type="point_cloud_processing_node" output="screen">
    <param name="subscriber" value="point_cloud_A"/>
    <param name="publisher" value="point_cloud_B"/>
    <param name="mode" value="outlierRemoval"/>
    <param name="msgSubscriber" value="zzzzzzzzz"/>
  </node>

  <node name="poin_cloud_processor3" pkg="point_cloud_processing" type="point_cloud_processing_node" output="screen">
    <param name="subscriber" value="point_cloud_B"/>
    <param name="publisher" value="point_cloud_R"/>
    <param name="mode" value="transform"/>
    <param name="msgSubscriber" value="zzzzzzzzz"/>
  </node>


</launch>
